### PR TITLE
Perf

### PR DESCRIPTION
**Pull request by TerribleDev**
* Cache Reflection calls to get Properties
* Short circuit properties over 8k and total log size over 16k

### CODE REVIEW

File: src/OKLogger/Parsing/ValueEscaper.cs [hunk 1]
```diff
@@ -23,9 +23,8 @@ public string Escape(string value)
             }
             foreach(var c in EscapedChars)
             {
-                value = value.Replace(c, string.Empty);
+                value = value.Replace(c, '\0'); // \0 is an ASCII value of 0 aka no value
```
Comment (adamwadesmith) [inline src/OKLogger/Parsing/ValueEscaper.cs]: I'm not sure how various programs will handle the \0.

Why not just replace " (double-quote) with ' (single quote). The need is quite specific - we don't require a generic escaping class.

File: src/OKLogger/ReflectionFactory.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        internal static IList<PropertyInfo> GetProperties<T>(T obj)
+        {
+            var objType = obj.GetType();
+            if(propertyInfos.TryGetValue(objType, out IList<PropertyInfo> propertyInfo))
+            {
+                return propertyInfo;
+            }
+            else
+            {
+                var runtimeProps = objType.GetRuntimeProperties().ToList();
```
Comment (adamwadesmith) [inline src/OKLogger/ReflectionFactory.cs:24]: Should the properties be sorted by name?

File: src/OKLogger/Parsing/ValueEscaper.cs (hunk 1 above)
Comment (TerribleDev) [reply src/OKLogger/Parsing/ValueEscaper.cs]: we can do that, as it stands now we string replace with an empty string so I wanted to maintain the same functionality.

File: src/OKLogger/ReflectionFactory.cs (hunk 2 above)
Comment (TerribleDev) [reply src/OKLogger/ReflectionFactory.cs:24]: I confirmed `GetRuntimeProperties()` gives us the properties in order of declaration. I think that is what we should stick with IMO.

**Review by cubsk — APPROVED** [review]

**Pull request merged**